controller/nodelabel_controller.go

```diff
@@ -371,7 +345,7 @@ func (r *ReconcileNodeLabel) reconcileVMs(namespacedName types.NamespacedName, p
 func (r *ReconcileNodeLabel) applyTagsToNodes(namespacedName types.NamespacedName, computeResource ComputeResource, node *corev1.Node, configOptions *ConfigOptions) ([]byte, error) {
 	log := r.Log.WithValues("node-label-operator", namespacedName)
 
-	changed := false
+	newLabels := map[string]*string{} // should allow for null JSON values
```

Comment:
I would try to make the Label concept a struct.
that would help organize some of the label specific validation, conversion and sanitizing on the the label object.
you can still use a map to make sure you don't have multiple times the same label if necessary.

Reply:
Ah okay! I need specifically a map for patching but I agree it could be more readable so I'll try using a Label struct.